tests/testsuite/rustc.rs

```diff
@@ -855,3 +855,18 @@ fn precedence() {
 "#]])
         .run();
 }
+
+#[cargo_test]
+fn build_with_duplicate_crate_types() {
+    let p = project().file("src/lib.rs", "").build();
+
+    // Test with duplicate crate types specified directly
+    p.cargo("rustc -v --crate-type staticlib --crate-type staticlib")
+        .with_stderr_data(str![[r#"
+[COMPILING] foo v0.0.1 ([ROOT]/foo)
+[RUNNING] `rustc --crate-name foo --edition=2015 src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type staticlib --emit=dep-info,link -C embed-bitcode=no -C debuginfo=2 -C split-debuginfo=unpacked --check-cfg 'cfg(docsrs,test)' --check-cfg 'cfg(feature, values())' -C metadata=991ef8c99cd510e0 -C extra-filename=-f0d97c77b50b1fb3 --out-dir [ROOT]/foo/target/debug/deps -L dependency=[ROOT]/foo/target/debug/deps`
```

Comment:
Assering the entire rustc argument list is a big fragile. Maybe this?
(`[..]` for wildcard matching.)

```suggestion
[RUNNING] `rustc [..]future-incompat --crate-type staticlib --emit[..]
```